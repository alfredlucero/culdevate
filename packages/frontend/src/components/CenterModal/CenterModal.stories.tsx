import React from "react";
import { storiesOf } from "@storybook/react";
import CenterModal from "./index";
import Text from "../Text";
import Heading from "../Heading";
import Button from "../Button";

const onHide = () => {
  console.log("Hide Center Modal!");
};

storiesOf("Common/Center Modal", module).add("Showing Example", () => (
  <CenterModal isShowing={true} onHide={onHide}>
    <Heading variant="h3">Center Modal Title</Heading>
    <Text variant="p" className="py-4">
      Center Modal Description
    </Text>
    <div className="flex justify-end pt-4">
      <Button type="button" variant="secondary" onClick={onHide} className="mr-2">
        Close
      </Button>
      <Button
        type="button"
        variant="primary"
        onClick={() => {
          console.log("Action!");
        }}
      >
        Action
      </Button>
    </div>
  </CenterModal>
));
