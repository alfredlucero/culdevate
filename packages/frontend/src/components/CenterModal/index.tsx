import React from "react";
import ReactDOM from "react-dom";
import { CommonProps } from "../commonProps";

export interface CenterModalProps extends CommonProps {
  isShowing: boolean;
  onHide: () => void;
  children: React.ReactNode;
}

const CenterModal: React.FC<CenterModalProps> = ({
  isShowing,
  onHide,
  children,
  testId = "",
  className = "",
  ...passThroughProps
}) => {
  if (!isShowing) {
    return null;
  }

  return ReactDOM.createPortal(
    <div
      className={`modal fixed w-full h-full top-0 left-0 flex items-center justify-center${
        className !== "" ? ` ${className}` : ""
      }`}
      {...(testId !== "" ? { "data-testid": testId } : {})}
      {...passThroughProps}
    >
      <div
        className="modal-overlay absolute w-full h-full bg-gray-900 opacity-50 z-40"
        data-testid="centerModalOverlay"
        onClick={onHide}
      />
      <div
        className="modal-container bg-white w-11/12 md:max-w-lg mx-auto rounded shadow-lg z-50 overflow-y-auto"
        aria-modal
        aria-hidden
        tabIndex={-1}
        role="dialog"
      >
        <div className="modal-content py-4 px-6">
          <div className="modal-header w-full flex justify-end">
            <svg
              className="fill-current h-6 w-6 text-gray-700"
              role="button"
              xmlns="http://www.w3.org/2000/svg"
              viewBox="0 0 20 20"
              data-dismiss="modal"
              aria-label="Close"
              onClick={onHide}
              data-testid="centerModalCloseButton"
            >
              <title>Close</title>
              <path d="M14.348 14.849a1.2 1.2 0 0 1-1.697 0L10 11.819l-2.651 3.029a1.2 1.2 0 1 1-1.697-1.697l2.758-3.15-2.759-3.152a1.2 1.2 0 1 1 1.697-1.697L10 8.183l2.651-3.031a1.2 1.2 0 1 1 1.697 1.697l-2.758 3.152 2.758 3.15a1.2 1.2 0 0 1 0 1.698z" />
            </svg>
          </div>
          {children}
        </div>
      </div>
    </div>,
    document.body,
  );
};

export default CenterModal;
