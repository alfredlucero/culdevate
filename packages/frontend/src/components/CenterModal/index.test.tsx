import React from "react";
import { render, fireEvent } from "@testing-library/react";
import CenterModal from "./index";

describe("<CenterModal />", () => {
  const centerModalTestId = "centerModalTestId";

  test("should render without error", () => {
    const { container } = render(
      <CenterModal isShowing={true} onHide={() => {}} testId={centerModalTestId} className="center-modal">
        Modal
      </CenterModal>,
    );

    expect(container).toMatchSnapshot();
  });

  test("should not exist when the center modal is not showing", () => {
    const { queryByTestId } = render(
      <CenterModal isShowing={false} onHide={() => {}} testId={centerModalTestId}>
        Modal
      </CenterModal>,
    );

    expect(queryByTestId(centerModalTestId)).toBeNull();
  });

  test("should call onHide callback when clicking the center modal's overlay", () => {
    const onHideMock = jest.fn();
    const { getByTestId } = render(
      <CenterModal isShowing={true} onHide={onHideMock} testId={centerModalTestId}>
        Modal
      </CenterModal>,
    );

    fireEvent.click(getByTestId("centerModalOverlay"));

    expect(onHideMock).toHaveBeenCalled();
  });

  test("should call onHide callback when clicking the close icon button", () => {
    const onHideMock = jest.fn();
    const { getByTestId } = render(
      <CenterModal isShowing={true} onHide={onHideMock} testId={centerModalTestId}>
        Modal
      </CenterModal>,
    );

    fireEvent.click(getByTestId("centerModalCloseButton"));

    expect(onHideMock).toHaveBeenCalled();
  });
});
