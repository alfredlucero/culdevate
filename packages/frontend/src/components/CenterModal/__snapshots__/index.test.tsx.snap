// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CenterModal /> should render without error 1`] = `<div />`;
